### DISCUSSION TITLE
Arrays of static types

### DISCUSSION THREAD

**Discussion by quixoticaxis** [dotnet/csharplang#5338] [Q&A] [answered]
The current draft does not seem to impose any restrictions on the type of array elements. Is it an oversight, or is it supposed to be that way?

I came to re-checking the spec after the discussion of [static abstract members in interfaces](https://github.com/dotnet/csharplang/issues/4436) moved to discussing the possibility of allowing static classes as generic arguments.

As [you can see](https://github.com/dotnet/csharplang/issues/4436#issuecomment-950613599), it's currently possible to instantiate the array of a static type, but not through normal language means.

**Comment by svick** [comment, marked as answer]
The element type of an array has to be a *type*, and [static classes are (mostly) not considered types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes?WT.mc_id=DOP-MVP-5002735#static-classes):

> A static class […] cannot be used as a type

So I think arrays of static types are forbidden by the spec, and I don't see anything that would need to change here.

**Reply by quixoticaxis** [reply]
I see, thank you.

**Reply by quixoticaxis** [reply]
The same [link](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes?WT.mc_id=DOP-MVP-5002735#static-classes) explicitely states: "Likewise, a static class cannot be used in an array type".
Quote added for visibility.
